x/evm/types/state_transition.go

```diff
@@ -73,6 +72,11 @@ func (st StateTransition) TransitionCSDB(ctx sdk.Context) (*ReturnData, error) {
 	// Clear cache of accounts to handle changes outside of the EVM
 	csdb.UpdateAccounts()
 
+	gasPrice := ctx.MinGasPrices().AmountOf(emint.DenomDefault)
+	if gasPrice.IsNil() {
+		panic("gas price cannot be nil")
```

Comment:
should we panic here or just return an error?

Reply:
good call 👍